UITests/SessionRestoreTests.swift

```diff
@@ -34,25 +34,44 @@ class SessionRestoreTests: KIFTestCase {
         // After triggering the restore, the session should look like this:
         //   about:home, page1, *page2*, page3
         // where page2 is active.
-        tester().tapViewWithAccessibilityIdentifier("url")
-        tester().clearTextFromAndThenEnterTextIntoCurrentFirstResponder("\(restoreURL!.absoluteString!)\n")
-        tester().waitForWebViewElementWithAccessibilityLabel("Page 2")
-        tester().tapViewWithAccessibilityLabel("Back")
-        tester().waitForWebViewElementWithAccessibilityLabel("Page 1")
-        tester().tapViewWithAccessibilityLabel("Back")
-        tester().waitForViewWithAccessibilityLabel("Top sites")
-        let canGoBack: Bool
+		EarlGrey().selectElementWithMatcher(grey_accessibilityID("url"))
+			.performAction(grey_tap())
+		EarlGrey().selectElementWithMatcher(grey_accessibilityID("address"))
+			.performAction(grey_typeText("\(restoreURL!.absoluteString!)\n"))
+		tester().waitForWebViewElementWithAccessibilityLabel("Page 2")
+        EarlGrey().selectElementWithMatcher(grey_accessibilityLabel("Back"))
+			.performAction(grey_tap())
+		
+		tester().waitForWebViewElementWithAccessibilityLabel("Page 1")
+        EarlGrey().selectElementWithMatcher(grey_accessibilityLabel("Back"))
+			.performAction(grey_tap())
+		let wentBack = GREYCondition(name: "Check browser went back", block: { _ in
+			var errorOrNil: NSError?
+			let matcher = grey_allOfMatchers(grey_accessibilityLabel("Top sites"),
+				grey_sufficientlyVisible())
+			EarlGrey().selectElementWithMatcher(matcher)
+				.assertWithMatcher(grey_notNil(), error: &errorOrNil)
+			let success = errorOrNil == nil
+			return success
+		}).waitWithTimeout(5)
+		GREYAssertTrue(wentBack, reason: "Didn't go back")
+		
+		let canGoBack: Bool
         do {
             try tester().tryFindingTappableViewWithAccessibilityLabel("Back")
             canGoBack = true
         } catch _ {
             canGoBack = false
         }
         XCTAssertFalse(canGoBack, "Reached the beginning of browser history")
-        tester().tapViewWithAccessibilityLabel("Forward")
-        tester().tapViewWithAccessibilityLabel("Forward")
-        tester().tapViewWithAccessibilityLabel("Forward")
-        tester().waitForWebViewElementWithAccessibilityLabel("Page 3")
+		
```

Comment:
ah, i think it's probably my xcode using tab instead of space for indentation, I'll fix this.